### DISCUSSION TITLE
Issue with installing sunbird portal  dependency.

### DISCUSSION THREAD

**Discussion by sachinsahu731** [Sunbird-Ed/Community#130] [Installation] [answered]
Below is the version in my machine

Window 10 
npm  -: 8.18.0
node -: v16.13.2
Angular CLI: 13.1.2

Added below variable in environment 
sunbird_environment      | local   |   string  |
   |  sunbird_instance         | sunbird |   string  |
   |  sunbird_default_channel  | sunbird |   string  |
   |  sunbird_default_tenant   | sunbird |   string  |

src/app/client folder when I try to install node dependency we got below error using `yarn install` 

`[5/5] Building fresh packages... [6/13] ⠄ @angular/cli [2/13] ⠄ @angular/cli [7/13] ⡀ @angular/cli [10/13] ⡀ iltorb error F:\Sunbirdfork\SunbirdEd-portal\src\app\client\node_modules\iltorb: Comman d failed. Exit code: 1 Command: node ./scripts/install.js || node-gyp rebuild Arguments: Directory: F:\Sunbirdfork\SunbirdEd-portal\src\app\client\node_modules\iltorb Output: info looking for cached prebuild @ C:\Users\Goodworks\AppData\Roaming\npm-cache\ _prebuilds\68a94b-iltorb-v2.4.5-node-v93-win32-x64.tar.gz http request GET https://github.com/nstepien/iltorb/releases/download/v2.4.5/ilt orb-v2.4.5-node-v93-win32-x64.tar.gz http 404 https://github.com/nstepien/iltorb/releases/download/v2.4.5/iltorb-v2.4 .5-node-v93-win32-x64.tar.gz WARN install No prebuilt binaries found (target=16.13.2 runtime=node arch=x64 li bc= platform=win32) F:\Sunbirdfork\SunbirdEd-portal\src\app\client\node_modules\iltorb>if not define d npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-g yp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "" re build ) gyp info it worked if it ends with ok gyp info using node-gyp@9.1.0 gyp info using node@16.13.2 | win32 | x64 gyp ERR! find Python gyp ERR! find Python checking Python explicitly set from command line or npm con figuration gyp ERR! find Python - "--python=" or "npm config get python" is "python2.7" gyp ERR! find Python - "python2.7" is not in PATH or produced an error gyp ERR! find Python Python is not set from environment variable PYTHON gyp ERR! find Python checking if "python3" can be used gyp ERR! find Python - "python3" is not in PATH or produced an error gyp ERR! find Python checking if "python" can be used gyp ERR! find Python - executable path is "C:\Python27\python.exe" gyp ERR! find Python - version is "2.7.8" gyp ERR! find Python - version is 2.7.8 - should be >=3.6.0 gyp ERR! find Python - THIS VERSION OF PYTHON IS NOT SUPPORTED gyp ERR! find Python checking if Python is C:\Users\Goodworks\AppData\Local\Prog rams\Python\Python39\python.exe gyp ERR! find Python - "C:\Users\Goodworks\AppData\Local\Programs\Python\Python3 9\python.exe" could not be run gyp ERR! find Python checking if Python is C:\Program Files\Python39\python.exe gyp ERR! find Python - "C:\Program Files\Python39\python.exe" could not be run gyp ERR! find Python checking if Python is C:\Users\Goodworks\AppData\Local\Prog rams\Python\Python39-32\python.exe gyp ERR! find Python - "C:\Users\Goodworks\AppData\Local\Programs\Python\Python3 9-32\python.exe" could not be run gyp ERR! find Python checking if Python is C:\Program Files\Python39-32\python.e xe gyp ERR! find Python - "C:\Program Files\Python39-32\python.exe" could not be ru n gyp ERR! find Python checking if Python is C:\Program Files (x86)\Python39-32\py thon.exe gyp ERR! find Python - "C:\Program Files (x86)\Python39-32\python.exe" could not be run gyp ERR! find Python checking if Python is C:\Users\Goodworks\AppData\Local\Prog rams\Python\Python38\python.exe gyp ERR! find Python - "C:\Users\Goodworks\AppData\Local\Programs\Python\Python3 8\python.exe" could not be run gyp ERR! find Python checking if Python is C:\Program Files\Python38\python.exe gyp ERR! find Python - "C:\Program Files\Python38\python.exe" could not be run gyp ERR! find Python checking if Python is C:\Users\Goodworks\AppData\Local\Prog rams\Python\Python38-32\python.exe gyp ERR! find Python - "C:\Users\Goodworks\AppData\Local\Programs\Python\Python3 8-32\python.exe" could not be run gyp ERR! find Python checking if Python is C:\Program Files\Python38-32\python.e xe gyp ERR! find Python - "C:\Program Files\Python38-32\python.exe" could not be ru n gyp ERR! find Python checking if Python is C:\Program Files (x86)\Python38-32\py thon.exe gyp ERR! find Python - "C:\Program Files (x86)\Python38-32\python.exe" could not be run gyp ERR! find Python checking if Python is C:\Users\Goodworks\AppData\Local\Prog rams\Python\Python37\python.exe gyp ERR! find Python - "C:\Users\Goodworks\AppData\Local\Programs\Python\Python3 7\python.exe" could not be run gyp ERR! find Python checking if Python is C:\Program Files\Python37\python.exe gyp ERR! find Python - "C:\Program Files\Python37\python.exe" could not be run gyp ERR! find Python checking if Python is C:\Users\Goodworks\AppData\Local\Prog rams\Python\Python37-32\python.exe gyp ERR! find Python - "C:\Users\Goodworks\AppData\Local\Programs\Python\Python3 7-32\python.exe" could not be run gyp ERR! find Python checking if Python is C:\Program Files\Python37-32\python.e xe gyp ERR! find Python - "C:\Program Files\Python37-32\python.exe" could not be ru n gyp ERR! find Python checking if Python is C:\Program Files (x86)\Python37-32\py thon.exe gyp ERR! find Python - "C:\Program Files (x86)\Python37-32\python.exe" could not be run gyp ERR! find Python checking if Python is C:\Users\Goodworks\AppData\Local\Prog rams\Python\Python36\python.exe gyp ERR! find Python - "C:\Users\Goodworks\AppData\Local\Programs\Python\Python3 6\python.exe" could not be run gyp ERR! find Python checking if Python is C:\Program Files\Python36\python.exe gyp ERR! find Python - "C:\Program Files\Python36\python.exe" could not be run gyp ERR! find Python checking if Python is C:\Users\Goodworks\AppData\Local\Prog rams\Python\Python36-32\python.exe gyp ERR! find Python - "C:\Users\Goodworks\AppData\Local\Programs\Python\Python3 6-32\python.exe" could not be run gyp ERR! find Python checking if Python is C:\Program Files\Python36-32\python.e xe gyp ERR! find Python - "C:\Program Files\Python36-32\python.exe" could not be ru n gyp ERR! find Python checking if Python is C:\Program Files (x86)\Python36-32\py thon.exe gyp ERR! find Python - "C:\Program Files (x86)\Python36-32\python.exe" could not be run gyp ERR! find Python checking if the py launcher can be used to find Python 3 gyp ERR! find Python - "py.exe" is not in PATH or produced an error gyp ERR! find Python gyp ERR! find Python ********************************************************** gyp ERR! find Python You need to install the latest version of Python. gyp ERR! find Python Node-gyp should be able to find and use Python. If not, gyp ERR! find Python you can try one of the following options: gyp ERR! find Python - Use the switch --python="C:\Path\To\python.exe" gyp ERR! find Python   (accepted by both node-gyp and npm) gyp ERR! find Python - Set the environment variable PYTHON gyp ERR! find Python - Set the npm configuration variable python: gyp ERR! find Python   npm config set python "C:\Path\To\python.exe" gyp ERR! find Python For more information consult the documentation at: gyp ERR! find Python https://github.com/nodejs/node-gyp#installation gyp ERR! find Python ********************************************************** gyp ERR! find Python gyp ERR! configure error gyp ERR! stack Error: Could not find any Python installation to use gyp ERR! stack     at PythonFinder.fail (C:\Users\Goodworks\AppData\Roaming\nvm\ v16.13.2\node_modules\npm\node_modules\node-gyp\lib\find-python.js:330:47) gyp ERR! stack     at PythonFinder.runChecks (C:\Users\Goodworks\AppData\Roaming \nvm\v16.13.2\node_modules\npm\node_modules\node-gyp\lib\find-python.js:159:21) gyp ERR! stack     at PythonFinder.<anonymous> (C:\Users\Goodworks\AppData\Roami ng\nvm\v16.13.2\node_modules\npm\node_modules\node-gyp\lib\find-python.js:228:18 ) gyp ERR! stack     at PythonFinder.execFileCallback (C:\Users\Goodworks\AppData\ Roaming\nvm\v16.13.2\node_modules\npm\node_modules\node-gyp\lib\find-python.js:2 94:16) gyp ERR! stack     at ChildProcess.exithandler (node:child_process:404:5) gyp ERR! stack     at ChildProcess.emit (node:events:390:28) gyp ERR! stack     at maybeClose (node:internal/child_process:1064:16) gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_p rocess:301:5) gyp ERR! System Windows_NT 10.0.19043 gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodej s\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild" gyp ERR! cwd F:\Sunbirdfork\SunbirdEd-portal\src\app\client\node_modules\iltorb gyp ERR! node -v v16.13.2`

**Comment by sharathkashyap** [comment, marked as answer]
@sachinsahu731 The issue your machine is facing is that most of the node_modules depends on implicit Python version installed. Looks like 
`find Python - "python2.7" is not in PATH or produced an error gyp ERR! find Python Python is not set from environment variable`
Python is not added into your environment.
Kindly add it in Path and environment variable.
cc: @luckynvdu

**Reply by sachinsahu731** [reply]
Thanks
